cime_config/acme/machines/template.lt_archive

```diff
@@ -1,177 +1,117 @@
-    }
-}
-
-#-----------------------------------------------------------------------------------------------
-# checkSta - check if short term archive is complete
-#-----------------------------------------------------------------------------------------------
-sub checkSta
-{
-    my $statusFile = shift;
-    my $staComplete = 0;
-
-    $logger->debug("In checkSta...");
-#
-# check if the short term archive completed successfully
-#
-    if( -f $statusFile )
-    {
-	open my $STASTATUS, "<", "$statusFile"  or die qq(lt_archive: unable to open $statusFile. Exiting...\n);
-	while( <$STASTATUS> )
-	{
-	    chomp $_;
-	        if( /^st_archive process complete/ )
-		{
-		    $staComplete = 1;
-		}
-	}
-	close( $STASTATUS );
-    }
-}
-
-
-#-----------------------------------------------------------------------------------------------
-# main
-#-----------------------------------------------------------------------------------------------
-sub main
-{
-    $ENV{'maxthrds'} = 1;
-
-    # First, get the configuration from every xml file.
-    %config = SetupTools::getAllResolved();
-
-
-    # Change to the case root
-    chdir($config{'CASEROOT'});
-    my $testlog = "$config{CASEROOT}/TestStatus.log";
-    if(-e $testlog)
-    {
-        open(TL, ">>$testlog");
-        print TL "lt_archive starting\n";
-        close(TL);
-    }
-    $logger->info("lt_archive starting");
+###############################################################################
+def parse_command_line(args):
+###############################################################################
+
+    CIME.utils.get_model()
+
+    parser = argparse.ArgumentParser()
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    args = parser.parse_args()
+
+    CIME.utils.handle_standard_logging_options(args)
+
+
+###############################################################################
+def _main_func():
+###############################################################################
+
+    parse_command_line(sys.argv)
+    
+    case = Case()
+
+    caseroot = case.get_value("CASEROOT")
+    testlog  = os.path.join(caseroot,"TestStatus.log")
+
+    # max number of threads needed by scripts
+    os.environ["maxthrds"] = 1
+
+    # document start
+    if os.path.isfile(testlog):
+        with open(testlog, "a") as fd:
+            fd.write("lt_archive starting %s" %(time.strftime("%Y-%m-%d %H:%M:%S")))
+    logger.info("lt_archive starting %s" %(time.strftime("%Y-%m-%d %H:%M:%S")))
+
+    # determine status of run and short term archiving
+    runComplete = False
+    statusfile = os.path.join(caseroot,"CaseStatus") 
+    if os.path.isfile(statusFile):
+        if 'run SUCCESSFUL' in open(statusFile).read():
+            runComplete = True
+
+    statusComplete = False
+    statusfile = os.path.join(caseroot,"stArchiveStatus")
+    if os.path.isfile(statusFile):
+        if 'st_archive process complete' in open(statusFile).read():
+            runComplete = True
```

Comment:
Done. Right. We need to add a fast long term archiving test. Jim is thinking about this as well.